### PR TITLE
Feature/4.1.1 mm filter and sort books

### PR DESCRIPTION
**Pull request by Kimara1904**
Implementation filter and order (sort) of books

### CODE REVIEW

File: fe-library-app/src/components/AppRouter/pages/HomePage.tsx [hunk 1]
```diff
@@ -7,40 +7,54 @@ import SearchBar from '../../SearchBar/SearchBar'
...
-
-  const getBooksPageAppend = (request: GetBooksRequest) => {
-    getBooks(request)
-      .then((response) => {
-        setBookList(prevList => [ ...prevList, ...response.data.Items ])
-        setIsResponseEmpty(request.PageNumber * request.PageLength >= response.data.TotalCount)
-      })
-      .catch(() => alert('Something went wrong!'))
-  }
+  const [ isThereBooks, setIsThereBooks ] = useState(false)
```
Comment (brankastevic) [inline fe-library-app/src/components/AppRouter/pages/HomePage.tsx:24]: Do you really need this additional state? 
You already have a bookList and you can use it. So, my suggestion is to use `bookList.length` for the condition.

File: fe-library-app/src/components/Filter/Filter.tsx [hunk 2]
```diff
@@ -1,5 +1,87 @@
...
+import styles from './Filter.module.css'
+
+interface FilterProps{
+  filterChange: (filterData: Filters) => void,
+  switchShowFilter: boolean,
+  hideFilter: () => void
+}
+
+const Filter = (props: FilterProps) => {
+  const initialState: Filters = {
```
Comment (brankastevic) [inline fe-library-app/src/components/Filter/Filter.tsx:13]: You can remove this const, there is no usage of this object in the code. As I see you always need one-by-one property.

You can do something like this instead:

`const [ desc, setDesc ] = useState('')
  const [ isbn, setIsbn ] = useState('')
  const [ authorFirstName, setAuthorFirstName ] = useState('')
  const [ authorLastName, setAuthorLastName ] = useState('')`

File: fe-library-app/src/components/SearchBar/SearchBar.tsx [hunk 3]
```diff
@@ -1,33 +1,53 @@
...
+  filterChange: (filterData: Filters) => void
+  orderChange: (orderData: string[]) => void
 }
 
 const SearchBar = (props: SearchBarProps) => {
   const [ showFilter, setShowFilter ] = useState(false)
+  const [ showOrder, setShowOrder ] = useState(false)
 
-  const onFilterHandler = () => {
+  const clickFilterHandler = () => {
```
Comment (brankastevic) [inline fe-library-app/src/components/SearchBar/SearchBar.tsx:19]: One more blog for reading: https://javascript.plainenglish.io/handy-naming-conventions-for-event-handler-functions-props-in-react-fc1cbb791364
If you are sometimes confused what is the best way to give a name to a function, there are some suggestion for best practices.

Comment (Kimara1904) [reply fe-library-app/src/components/SearchBar/SearchBar.tsx:19]: Thanks

**Review by brankastevic — APPROVED** [review]

**Pull request merged**